amlb/frameworks/definitions.py

```diff
@@ -138,19 +139,19 @@ def _add_default_params(framework):
         framework.params = Namespace.dict(framework.params)
 
 
-def _add_default_image(framework: Namespace, config: Namespace):
+def _add_default_image(framework: Namespace, config: Namespace, props=None):
```

Comment:
```suggestion
def _add_default_image(framework: Namespace, config: Namespace, props: Optional[List[str]] = None):
```
will have to also import `Optional` :/

Reply:
added the typing annotation, but in general I think we can be relaxed with private methods.
@PGijsbers using this comment to suggest some guidelines.
Typing annotations can be useful to define the contract of public methods/functions, much less for private ones as those are susceptible to have their contract changed at any time.
I think it's much more important to keep all method signatures small (2-3 positional params max) to make them easy to reason about, and intuitive.
Type system should be here to help, not to make things look more complex, or worse, add unnecessary time-wasting constraints (having written a lot of code in both strongly typed languages and loosely types ones, I can tell that both have their benefits depending on the context, the abstration level...  and we should not be religious about it).
Same with testing: many unit tests currently test private methods. This should be avoided: private methods are allowed to change, and then we spend a lot of time fixing those tests. In Java world for example, you never write UTs for private methods. In the same way, our tests should focus on scenarios on the public functions/methods.